src/org/labkey/test/WebDriverWrapper.java

```diff
@@ -2872,7 +2893,11 @@ public void clickAndWait(Locator l)
 
     public void openLinkInNewWindow(WebElement link)
     {
+        int winCount = getDriver().getWindowHandles().size();
         link.sendKeys(Keys.chord(WebDriverUtils.MODIFIER_KEY, Keys.ENTER));
+        if (getDriver().getWindowHandles().size() < (winCount + 1))
+            throw new IllegalStateException("Link did not open new window in tab.");
+
         switchToWindow(1);
         waitForDocument();
```

Comment:
`switchToWindow` already waits for the new window. However, this should be updated to switch to the new window instead of always using index 1.
```suggestion
        int winCount = getDriver().getWindowHandles().size();
        link.sendKeys(Keys.chord(WebDriverUtils.MODIFIER_KEY, Keys.ENTER));

        switchToWindow(winCount);
        waitForDocument();
```

Reply:
So I'm trying to test all anchor tags that could function like links, whether they have a href or a click event handler attached later. And there are many anchor tags on our pages that are not links. So I'm using this function to determine if an anchor tag is a link by trying to click on it and see if a new window pops up. If so, it's a real link. If not, it throws which is caught in the crawler test and it moves on to the next anchor tag. I incorporated your suggestion above, but also created a new function, openLinkInNewWindowOrThrow more designed for this use case.